search/src/main/java/artic/edu/search/DefaultSuggestionAdapter.kt

```diff
@@ -0,0 +1,84 @@
+package artic.edu.search
+
+import android.support.v4.content.ContextCompat
+import android.view.View
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.RequestOptions
+import com.fuzz.rx.bindTo
+import com.fuzz.rx.disposedBy
+import com.jakewharton.rxbinding2.widget.text
+import edu.artic.adapter.AutoHolderRecyclerViewAdapter
+import edu.artic.adapter.BaseViewHolder
+import kotlinx.android.synthetic.main.layout_cell_amenity.view.*
+import kotlinx.android.synthetic.main.layout_cell_header.view.*
+import kotlinx.android.synthetic.main.layout_cell_suggested_keyword.view.*
+import kotlinx.android.synthetic.main.layout_cell_suggested_map_object.view.*
+
+/**
+ * Adapter for loading default search suggestions
+ * @author Sameer Dhakal (Fuzz)
+ */
+class DefaultSuggestionAdapter : AutoHolderRecyclerViewAdapter<SearchBaseCellViewModel>() {
+    override fun View.onBindView(item: SearchBaseCellViewModel, position: Int) {
+        when (item) {
+            is TextCellViewModel -> {
+                item.text
+                        .bindTo(suggestedKeyword.text())
+                        .disposedBy(item.viewDisposeBag)
+            }
+            is HeaderCellViewModel -> {
+                item.text
+                        .bindTo(headerText.text())
+                        .disposedBy(item.viewDisposeBag)
+            }
+            is CircularCellViewModel -> {
+                item.imageUrl
+                        .subscribe {
+                            Glide.with(this)
+                                    .load(it)
+                                    .apply(RequestOptions.circleCropTransform())
+                                    .into(circularImage)
+                        }
+                        .disposedBy(item.viewDisposeBag)
+            }
+            is AmenitiesCellViewModel -> {
+                item.resource.subscribe { iconDrawable ->
+                    icon.setImageDrawable(ContextCompat.getDrawable(icon.context, iconDrawable))
+                }.disposedBy(item.viewDisposeBag)
+            }
+        }
+    }
+
+    override fun getLayoutResId(position: Int): Int {
+        val item = getItem(position)
+        return when (item) {
+            is HeaderCellViewModel -> {
```

Comment:
remove braces and return from these statements to clean up the `when()` into more concise.